### ISSUE TITLE
Ansible fails to process facter_* variables with facter version 4 

### ISSUE THREAD

**Issue by kotso** [ansible/ansible#81219]
### Summary

I'm using `facter_*` variables in my playbooks.

Im using playbooks on CentOS 7, Rocky and AlmaLinux 8 and 9 and Debian 11 and 12.

I have issue that `facter_*` variables are not set from setup in Alma/Rocky 9 and Debian 12.

This variables are set correctly on CentOS7, Almalinux/Rocky 8 and Debian 11.

What I have identified, working systems have facter  major version 3.x and non working systems have facter major version 4.x

If I call separately `community.general.facter`, it show all facts as it should, but issue is that `facter_*` variables are not set by setup, and I relay on this feature on so many playbooks. 

### Issue Type

Bug Report

### Component Name

setup

### Ansible Version

```console
$ ansible --version
ansible [core 2.15.1]
  config file = /Users/user/Documents/Development/project/Automation/Ansible/scripts/common/ansible.cfg
  configured module search path = ['/Users/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/lib/python3.11/site-packages/ansible
  ansible collection location = /Users/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin//ansible
  python version = 3.11.4 (main, Jun  7 2023, 00:34:59) [Clang 14.0.3 (clang-1403.0.22.14.1)] (/opt/homebrew/opt/python@3.11/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
```

### OS / Environment

Working on: 

CentOS 7, Facter 3
AlmaLinux 8, Facter 3
Debian 11, Facter 3

Not working:

AlmaLinux 9, Facter 4
Debian 12, Facter 4

### Steps to Reproduce

```yaml (paste below)
ansible server  -m setup 
ansible server  -m facter
```

### Expected Results

expected that setup would set facter results with facter_ prefix.

### Actual Results

```console
No facter variables set from facter executable,
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by kotso** [comment]
after some digging around, seems root cause is https://github.com/ansible/ansible/pull/80645 

after installing puppet on systems behaviour changed, and facts has been set, now experiencing issue with delegating this facts to other hosts, but it may be cache issue, doing some research on this.

**Comment by kotso** [comment]
ok, issue identified.

if facter 4.3 installed without puppet (may be other facter 4.x also have same behaviour but both RHEL 9 and Debian 12 is shipped with 4.3 this moment)

facter is not processed by setup (but processed with `community.general.facter`)

workaround for issue is installing puppet on server, after this expected behaviour is restored.

**Comment by jborean93** [comment]
The change you've mentioned https://github.com/ansible/ansible/pull/80645 just adds a fallback in case the original facter call failed, in this case it just retries the same facter call but without the `--puppet` argument. This change will still run the original path set beforehand so I'm not sure how that would have caused your problem. What happens if you manually call `facter --puppet --json` or `facter --json` outside of Ansible on the affected hosts.

**Comment by kotso** [comment]
@jborean93 since I have fixed issue by installing puppet on all servers, now I can not provide output.

But while debugging issue, I was trying to manually run commands you stated above, but only `facter --json` and it was giving warning that puppet not found and still showing valid json that I parsed with `jq` and verified that keys I'm looking where included.

I have not tried `facter --puppet --json`.
